src/proposals/20250109/SparkEthereum_20250109.t.sol

```diff
@@ -0,0 +1,926 @@
+
+        executeAllPayloadsAndBridges();
+
+        assertEq(proxy.hasRole(proxy.CONTROLLER(), Base.ALM_CONTROLLER),     false);
+        assertEq(proxy.hasRole(proxy.CONTROLLER(), BASE_NEW_ALM_CONTROLLER), true);
+    }
+
+    function test_BASE_AaveOnboardingIntegration() public onChain(ChainIdUtils.Base()) {
+        executeAllPayloadsAndBridges();
+
+        ForeignController controller = ForeignController(BASE_NEW_ALM_CONTROLLER);
+        
+        IERC20 usdc  = IERC20(Base.USDC);
+        IERC20 ausdc = IERC20(BASE_ATOKEN_USDC);
+
+        // Use a realistic numbers to check the rate limits
+        uint256 usdcAmount = 5_000_000e6;
+
+        // Use deal2 for USDC because storage is not set in a common way
+        deal2(Base.USDC, Base.ALM_PROXY, usdcAmount);
+
+        assertEq(usdc.balanceOf(Base.ALM_PROXY),  usdcAmount);
+        assertEq(ausdc.balanceOf(Base.ALM_PROXY), 0);
+
+        vm.startPrank(Base.ALM_RELAYER);
+
+        controller.depositAave(BASE_ATOKEN_USDC, usdcAmount);
+
+        assertEq(usdc.balanceOf(Base.ALM_PROXY),  0);
+        assertEq(ausdc.balanceOf(Base.ALM_PROXY), usdcAmount);
+
+        controller.withdrawAave(BASE_ATOKEN_USDC, usdcAmount);
+
+        assertEq(usdc.balanceOf(Base.ALM_PROXY),  usdcAmount);
+        assertEq(ausdc.balanceOf(Base.ALM_PROXY), 0);
+    }
+
+    function test_BASE_AaveRateLimits() public onChain(ChainIdUtils.Base()) {
+        executeAllPayloadsAndBridges();
+
+        MainnetController controller = MainnetController(BASE_NEW_ALM_CONTROLLER);
+        IRateLimits rateLimits       = IRateLimits(Base.ALM_RATE_LIMITS);
+        
+        IERC20 usdc  = IERC20(Base.USDC);
+        IERC20 ausdc = IERC20(BASE_ATOKEN_USDC);
+
+        // Use deal2 for USDC because storage is not set in a common way
+        deal2(Base.USDC, Base.ALM_PROXY, 50_000_000e6);
+
+        vm.startPrank(Base.ALM_RELAYER);
+
+        // USDC
+
+        bytes32 usdcDepositKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_AAVE_DEPOSIT(),
+            address(ausdc)
+        );
+        bytes32 usdcWithdrawKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_AAVE_WITHDRAW(),
+            address(ausdc)
+        );
+
+        assertEq(rateLimits.getCurrentRateLimit(usdcDepositKey), 50_000_000e6);
+        assertEq(usdc.balanceOf(Base.ALM_PROXY),                 50_000_000e6);
+        assertEq(ausdc.balanceOf(Base.ALM_PROXY),                0);
+
+        controller.depositAave(address(ausdc), 50_000_000e6);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdcDepositKey), 0);
+        assertEq(usdc.balanceOf(Base.ALM_PROXY),                 0);
+        assertEq(ausdc.balanceOf(Base.ALM_PROXY),                50_000_000e6);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdcWithdrawKey), type(uint256).max);
+
+        // All limits should be reset in 2 days + 1 (rounding)
+        skip(2 days + 1);
```

Comment:
Like other rate limit test comments, should have an intermediate assertion here to make sure rate limit is increasing at expected rate

Reply:
updated